src/lsm/enforcement.bpf.c

```diff
@@ -59,48 +66,53 @@ int BPF_PROG(bprm_check_security, struct linux_binprm *bprm)
 SEC("lsm/file_open")
 int BPF_PROG(file_open, struct file *file)
 {
-	//bpf_printk("lsm_hook: file: file_open\n");
-
-	event *e = bpf_ringbuf_reserve(&events, sizeof(*e), 0);
+    event *e = bpf_ringbuf_reserve(&events, sizeof(*e), 0);
     if (!e) {
-        bpf_printk("Faild ringbuf_reserve");
+        bpf_printk("Failed ringbuf_reserve");
         return 0;
-    }    
-    
+    }
+
     int ret = init_context(e);
     if (ret < 0) {
         bpf_ringbuf_discard(e, 0);
         return 0;
     }
-    
-    if(bpf_d_path(&file->f_path, e->data.path, sizeof(e->data.path)) < 0){
+
+    if (bpf_d_path(&file->f_path, e->data.path, sizeof(e->data.path)) < 0) {
         bpf_printk("Failed to get file path");
     }
 
     get_process_path(e->data.source, sizeof(e->data.source));
 
     __u32 flags = match_policy(POLICY_FILE, e->data.path);
 
-    // If there is no policy, allow
     if (!flags) goto clear;
 
     ret = 0;
-    // If you need to explicitly allow (whitelist | access list), it is denied by default.
     __u8 mode = flags & POLICY_ALLOW;
     if (mode) e->retval = -1;
+
     
-    if (
-        (flags & POLICY_FILE_READ)
-        || (flags & POLICY_FILE_WRITE)
-        || (flags & POLICY_FILE_EXEC)
-    ) ret -= 1;
+    if ((flags & POLICY_FILE_READ) && ((file->f_flags & O_WRONLY) == 0)) {
+        bpf_printk("block read at %s %d \n", e->data.path, flags);
+        ret -= 1;
+    }
```

Comment:
read block 을 하는데 write only가 아니다를 확인하는 이유가 궁금합니다

Reply:
해당 플래그에 따라서 읽기 전용 혹은 읽기-쓰기 플래그의 유무를 체크한건데, 이 부분이 조금 애매한게 read못하는데 write를 할 수 있다는게 상식적으로 이해는 안돼서 현재 고민중입니다.

=>

```suggestion
 if ((flags & POLICY_FILE_READ) && ((file->f_flags & (O_RDONLY | O_RDWR)) != 0)) {
    bpf_printk("block read at %s %d \n", e->data.path, flags);
    ret -= 1;
}
```
이게 더 올바른 차단 방식일 수도 있겠네요. 의견 감사합니다 👍

Reply:
해당 방식이 특정 모드에 대해서만 탐지를 하는 로직이라, 정상적으로 작동을 하지 않아 다시 돌리도록 하겠습니다